awx/main/signals.py

```diff
@@ -684,7 +684,9 @@ def save_user_session_membership(sender, **kwargs):
         return
     if UserSessionMembership.objects.filter(user=user_id, session=session).exists():
         return
-    UserSessionMembership(user_id=user_id, session=session, created=timezone.now()).save()
+    # check if user_id from session has an id match in User before saving
+    if User.objects.filter(id=int(user_id)).exists():
+        UserSessionMembership(user_id=user_id, session=session, created=timezone.now()).save()
```

Comment:
seems like you should go ahead and indent the rest of this method. If the user doesn't exist, I don't think we want to go expiring session memberships for that deleted user.